pallets/liquid-staking/src/lib.rs

```diff
@@ -454,61 +485,41 @@ pub mod pallet {
             Ok(().into())
         }
 
-        /// Do settlement for matching pool.
-        ///
-        /// The extrinsic does two things:
-        /// 1. Update exchange rate
-        /// 2. Calculate the imbalance of current matching state and send corresponding operations to
-        /// relay-chain.
-        #[pallet::weight(<T as Config>::WeightInfo::settlement())]
+        /// Update staking_ledger for updating exchange rate in next era
+        #[pallet::weight(<T as Config>::WeightInfo::update_staking_ledger())]
         #[transactional]
-        pub fn settlement(
+        pub fn update_staking_ledger(
             origin: OriginFor<T>,
-            #[pallet::compact] bonding_amount: BalanceOf<T>,
-            #[pallet::compact] unbonding_amount: BalanceOf<T>,
+            derivative_index: DerivativeIndex,
+            staking_ledger: StakingLedger<T::AccountId, BalanceOf<T>>,
         ) -> DispatchResultWithPostInfo {
             Self::ensure_origin(origin)?;
 
-            let relaychain_blocknumber = T::RelayChainBlockNumberProvider::current_block_number();
-            ensure!(
-                relaychain_blocknumber.saturating_sub(Self::last_settlement_time())
-                    >= T::EraLength::get(),
-                Error::<T>::SettlementLocked
-            );
-
-            let (bond_amount, rebond_amount, unbond_amount) =
-                Self::matching_pool().matching(unbonding_amount)?;
-
-            if bonding_amount.is_zero() && unbonding_amount.is_zero() {
-                Self::do_bond(bond_amount, RewardDestination::Staked)?;
-            } else {
-                Self::do_bond_extra(bond_amount)?;
+            if !StakingLedgers::<T>::contains_key(&derivative_index) {
+                return Ok(().into());
             }
 
-            Self::do_unbond(unbond_amount)?;
-            Self::do_rebond(rebond_amount)?;
-
-            Self::do_update_exchange_rate(bonding_amount)?;
+            Self::do_update_ledger(derivative_index, |ledger| {
+                // TODO: validate staking_ledger using storage proof
+                ensure!(
+                    ledger.unlocking == staking_ledger.unlocking
+                        && ledger.active == staking_ledger.active,
```

Comment:
This looks not right, because the bond type is `RewardDestination::Staked`, this will increase`ledger.active`

Reply:
you are right, I thought I was checking another value